dash-pipeline/bmv2/dash_inbound.p4

```diff
@@ -36,13 +35,15 @@ control inbound(inout headers_t hdr,
         ConntrackOut.apply(hdr, meta);
 #endif //PNA_CONNTRACK
 
-        vxlan_encap(hdr,
-                    meta.encap_data.underlay_dmac,
-                    meta.encap_data.underlay_smac,
-                    meta.encap_data.underlay_dip,
-                    meta.encap_data.underlay_sip,
-                    hdr.ethernet.dst_addr,
-                    meta.encap_data.vni);
+        tunnel_encap(hdr,
+                     meta,
+                     meta.encap_data.overlay_dmac,
```

Comment:
Since you are now passing the complete meta struct to tunnel_encap, can we simply remove all the meta.encap_data* fields from the argument list and use meta.encap_data.* fields directly inside the control block?

Reply:
I think it's still more readable with the nested struct.